internal_transfers/actions/src/accounting.ts

```diff
@@ -64,28 +64,69 @@ export async function simulateSolverSolution(
     return null;
   }
 
-  const commonSimulationParams = {
-    contractAddress: SETTLEMENT_CONTRACT_ADDRESS,
-    sender: solverAddress,
-    value: "0",
-    blockNumber: competition.simulationBlock,
-  };
-  const simFull = await simulator.simulate({
-    ...commonSimulationParams,
-    callData: competition.fullCallData,
-  });
-  const simReduced = await simulator.simulate({
-    ...commonSimulationParams,
-    callData: competition.reducedCallData,
+  const { full, reduced } = await simulateBoth(simulator, {
+    full: competition.fullCallData,
+    reduced: competition.reducedCallData,
+    common: {
+      contractAddress: SETTLEMENT_CONTRACT_ADDRESS,
+      sender: solverAddress,
+      value: "0",
+      blockNumber: competition.simulationBlock,
+    },
   });
+
   return {
     txHash: transaction.hash,
     winningSettlement: competition,
-    full: simFull,
-    reduced: simReduced,
+    full,
+    reduced,
   };
 }
 
+interface commonSimulationParams {
+  contractAddress: string;
+  sender: string;
+  value: string;
+  blockNumber: number;
+}
+
+interface SettlementSimulationParams {
+  full: string;
+  reduced: string;
+  common: commonSimulationParams;
+}
+
+interface SimulationPair {
+  full: SimulationData;
+  reduced: SimulationData;
+}
+async function simulateBoth(
+  simulator: TransactionSimulator,
+  params: SettlementSimulationParams,
+  numAttempts: number = 2
+): Promise<SimulationPair> {
+  let attempts = 0;
+  while (attempts < numAttempts) {
+    try {
+      return {
+        full: await simulator.simulate({
+          ...params.common,
+          callData: params.full,
+        }),
+        reduced: await simulator.simulate({
+          ...params.common,
+          callData: params.reduced,
+        }),
+      };
```

Comment:
Nit: I think we could wrap this in a for loop to reduce the amount of duplicated code. What's your take on that?

Reply:
I'm not sure I understand... there is no duplicated code here. It's as compact as it possibly can be.

Reply:
I believe you're calling the same function twice, just with different parameters. Therefore I assumed you could create a list of parameters and call the function for each parameter.